lib/googleauth.rb

```diff
@@ -33,3 +33,50 @@
 require 'googleauth/default_credentials'
 require 'googleauth/user_authorizer'
 require 'googleauth/web_user_authorizer'
+
+module Google
+  # Module Auth provides classes that provide Google-specific authorization
+  # used to access Google APIs.
+  module Auth
+    # On March 31, 2019, set supported version to 2.4 and recommended to 2.6.
+    # Thereafter, follow the MRI support schedule: supported means non-EOL,
+    # and recommended means in normal (rather than security) maintenance.
+    # See https://www.ruby-lang.org/en/downloads/branches/
+    ##
+    # Minimum "supported" Ruby version (non-EOL)
+    # @private
+    #
+    SUPPORTED_VERSION_THRESHOLD = '1.9'.freeze
+    ##
+    # Minimum "recommended" Ruby version (normal maintenance)
+    # @private
+    #
+    RECOMMENDED_VERSION_THRESHOLD = '2.4'.freeze
+    ##
+    # Check Ruby version and emit a warning if it is old
+    # @private
+    #
+    def self.warn_on_old_ruby_version
```

Comment:
Look at the current code in google-cloud-core for updated code that includes an environment variable for suppressing the warning. (I also realize I need to update the version constants over there to reflect what we now support...)